test/integration/framework/etcd.go

```diff
@@ -136,13 +153,65 @@ func RunCustomEtcd(dataDir string, customFlags []string, output io.Writer) (url
 	}
 	args = append(args, customFlags...)
 	cmd := exec.CommandContext(ctx, etcdPath, args...)
-	if output == nil {
-		cmd.Stdout = os.Stdout
-		cmd.Stderr = os.Stderr
-	} else {
-		cmd.Stdout = output
-		cmd.Stderr = output
-	}
+
+	// Always filter the etcd output. This allows us to get rid of known harmless messages and (more importantly)
+	// to detect the "listen tcp 127.0.0.1:37803: bind: address already in use" error.
+	reader, writer := io.Pipe()
+	cmd.Stdout = writer
+	cmd.Stderr = writer
+	var portInUse atomic.Bool
```

Comment:
Actually checking this is missing. While writing a test for it I noticed that the existing code which connects to the port doesn't detect that *someone else* took over the port -> more work needed, WIP.